### ISSUE TITLE
dialog: handle label automatically

### ISSUE THREAD

**Issue by ryanflorence** [reach/reach-ui#20]
Dialogs should have `aria-label` or `aria-labelled-by`, thinking we could help people out by requiring you render a `<DialogLabel>This is the label</DialogLabel>`.

If a dialog renders and doesn't find a `DialogLabel` inside, it could warn in the console.

Internally it could handle generating the ids for the developer and hook up the `aria-labelled-by`.

**Comment by giuseppeg** [comment]
I like the `accessibilityLabel` prop approach of React Native[ for Web]

**Comment by everdimension** [comment]
`<DialogLabel />` component kinda implies that you can render arbitrary markup inside of it, which, as I understand, is not the idea.

**Comment by chancestrickland** [comment]
`accessibilityLabel` Feels like a nice approach here to adopt as a standard across the project. Nice bonus that it will be familiar for native folks.

**Comment by kylemh** [comment]
Was curious if leveraging AlertDialogLabel is valid within a regular Dialog component?
